### ISSUE TITLE
Improved navigation between products / versions

### ISSUE THREAD

**Issue by eriknson** [MetaMask/mm-docs-v2#97]
Let's add a filter for (1) product and (2) version above the left-side-nav instead of our current tabs in the top bar to:
- Visually connect the filtering with the left-side-nav.
- Create a surface where we can briefly introduce each of our product via the subtitles.
- Support versioning.
- Scale smoothly if we want to add more top-level filters.

<img width="1651" alt="Docs navigation-closeup" src="https://user-images.githubusercontent.com/34306844/229746300-e715a5fe-49e6-4693-a55f-c8beb9a3d4f2.png">

**Overview**
<img width="1651" alt="Docs-navigation" src="https://user-images.githubusercontent.com/34306844/229732846-33fe18bf-4ec9-4ed3-9b55-d87f8ed34c68.png">

**Comment by vandan** [comment]
The Product Filter + Version looks nice. It keeps the top-nav clean. 

In this arrangement, do you have thoughts about where we would put an intro or general concepts?
I'm thinking we could still default to Wallet API, which could include general guidelines and prominent references to the other products.

**Comment by eriknson** [comment]
@vandan exactly, I'm thinking we would have a default choice which would include that kind of content
